internal/corazawaf/rule.go

```diff
@@ -831,3 +872,117 @@ func minPhase(v variables.RuleVariable) types.RulePhase {
 
 	return types.PhaseUnknown
 }
+
+// generateChainMatches generates matched chains based on the matchedValues. The latter provides all the variables that matched and their depth in the chain
+// generateChainMatches splits them into variables chains matches.
+// E.g. REQUEST_URI (chainLevel 0), REQUEST_URI (chainLevel 1), REQUEST_HEADERS (chainLevel 1), REQUEST_BODY (chainLevel 2), REQUEST_HEADERS (chainLevel 2)
+// REQUEST_URI - REQUEST_URI - REQUEST_BODY
+// REQUEST_URI - REQUEST_URI - REQUEST_HEADERS
+// REQUEST_URI - REQUEST_HEADERS - REQUEST_BODY
+// REQUEST_URI - REQUEST_HEADERS - REQUEST_HEADERS
+func generateChainMatches(tx *Transaction, matchedValues []types.MatchData, currentDepth int, buildingMatchedChain []types.MatchData, matchedChainsResult *[][]types.MatchData) {
```

Comment:
Is it being used outside multiphase? If so, would you please move it to a rule_multiphase.go?